### DISCUSSION TITLE
Passing dynamic property to queryFn for server filtering

### DISCUSSION THREAD

**Discussion by beard7** [TanStack/query#6250] [Q&A]
I'm trying to pass a parameter to the queryFn that can be used in the query string used to get data from my API. I thought it would simply be a case of using a dependent variable in the query key:
_"Adding dependent variables to your query key will ensure that queries are cached independently, and that any time a variable changes, queries will be refetched automatically"_

So I'm using a queryKey like this: ['assets', this.filter]

However, when the value of this.filter is changed (it's linked to a search field), the query is not being refetched and if I manually refetch() the data, the new value of this.filter still isn't applied.

What am I doing wrong?

**Comment by TkDodo** [comment]
Nothing, this should work. Please show a reproduction
